### PR TITLE
Adding documentation for Windows Customization

### PR DESCRIPTION
**Pull request by joadoumie**
## Summary of the pull request

1. Added guidelines into CONTRIBUTING.md for folks who'd like to suggest new settings/features to be incorporated into Windows Customization. 
2. Added a new issue template specific to feature requests for Windows Customization. Every community PR for a new feature in Windows Customization must reference an issue that has already been reviewed and signed-off by the team.

## Detailed description of the pull request / Additional comments

Check the file changes for more details -- there aren't an overwhelming amount 😊. 

## PR checklist
- [ ] Closes #xxx
- [ ] Tests added/passed
- [x] Documentation updated

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -112,6 +112,14 @@ Bugs and tasks are obviously the easiest to get started with, but don't feel afr
 
 Generally, we like to assign issues that generally belong to somebody's area of expertise to the team member that owns that area. This doesn't mean the community can't jump in -- they should reach out and have a chat with the assignee to see if it'd okay to take. If an issue was assigned more than a month ago, there's a good chance it's fair game to try yourself.
 
+### Contributing to Windows Customization
+
+If you'd like to suggest a new feature/enhancement in Windows Customization, **you must first file an issue with the provided `Windows Customization` template**. This will help us understand what you're looking for and why, and will help us ensure that the feature is something that we can support in the long run. For changes that rely on registry key behaviors that are undocumented, we may first have to review with internal stakeholders how to support the desired functionality and may not be able to support them in the long run. In these cases, we may have to modify your PR with a different approach after chatting with the internal teams.
```
Comment (krschau) [inline CONTRIBUTING.md:117]: Suggestion: link the reference to the template to the "open issue" link
`https://github.com/microsoft/devhome/issues/new?template=feature_request_windows_customization.yml`

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -112,6 +112,14 @@ Bugs and tasks are obviously the easiest to get started with, but don't feel afr
...
+
+We will not accept or review PRs that add new features to Windows Customization without an associated issue that follows the `Windows Customization` template.
+
+For bug fixes, simply file an issue with the appropriate label and we'll take a look. If you are able to fix the bug, please indicate that in the issue and we'll be happy to review your PR.
```
Comment (krschau) [inline CONTRIBUTING.md:121]: I don't believe community members can add/edit labels.

Comment (joadoumie) [reply CONTRIBUTING.md:121]: Good call. Will change.

File: .github/ISSUE_TEMPLATE/Feature_Request_Windows_Customization.yml [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+- id: minwindowsversion
+  type: input
+  attributes:
+    label: Minimum Supported Windows Version
+    placeholder: "10.0.23400.0"
+    description: What is the minimum version of Windows that should support this feature?
+  validations:
+    required: true
+
+- id: maxwindowsversion
```
Comment (dhoehna) [inline .github/ISSUE_TEMPLATE/Feature_Request_Windows_Customization.yml:17]: Why would someone want to enter a max version?  I would want the change to always work on every version in the future.

File: .github/ISSUE_TEMPLATE/Feature_Request_Windows_Customization.yml [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+- id: registrykeys
+  type: textarea
+  attributes:
+    label: Registry Keys
+    placeholder: What registry keys are needed to enable this feature? Please provide the full path and value. If not applicable, please enter "None".
+  validations:
+    required: true 
+
+- id: ispublic
+  type: dropdown
```
Comment (dhoehna) [inline .github/ISSUE_TEMPLATE/Feature_Request_Windows_Customization.yml:35]: What is the purpose of asking for documentation?

File: CONTRIBUTING.md (hunk 1 above)
Comment (dhoehna) [inline CONTRIBUTING.md:117]: The .yml uses `feature/improvement` but this doc uses `feature/enhancement`  should the two match?

Comment (dhoehna) [inline CONTRIBUTING.md:117]: NIT: long `term` not long `run`

Comment (dhoehna) [inline CONTRIBUTING.md:117]: `undocumented, we may first have` may?  This sounds like we would always consult internal stakeholders in this case.

File: .github/ISSUE_TEMPLATE/Feature_Request_Windows_Customization.yml (hunk 3 above)
Comment (joadoumie) [reply .github/ISSUE_TEMPLATE/Feature_Request_Windows_Customization.yml:17]: If there is a known maximum version where the change no longer is applicable or no longer works. We may have to work within these constraints.

File: .github/ISSUE_TEMPLATE/Feature_Request_Windows_Customization.yml (hunk 4 above)
Comment (joadoumie) [reply .github/ISSUE_TEMPLATE/Feature_Request_Windows_Customization.yml:35]: We need to confirm that the behavior is publicly documented before we approve the PR. There may be requests to enable specific behaviors that are well known, but not publicly documented and we'd like to ensure that we are not erroneously assuming some behavior that works now will work forever if it is not publicly documented and available by Microsoft.

File: CONTRIBUTING.md (hunk 1 above)
Comment (joadoumie) [reply CONTRIBUTING.md:117]: good catch. fixed.

[the remark this answers is not in the document]
Comment (joadoumie) [reply CONTRIBUTING.md:117]: We may first have to create private wrappers. We 100% want to make sure we are giving credit where credit is due, so we should take the stance that we would help get them past the finish line but that we may have to create private wrappers after discussing with internal stakeholders.

File: .github/ISSUE_TEMPLATE/Feature_Request_Windows_Customization.yml (hunk 4 above)
Comment (dhoehna) [reply .github/ISSUE_TEMPLATE/Feature_Request_Windows_Customization.yml:35]: I see.  You're equating public documentation to supported behavior.  Searching for public documentation is our job.  Other developers should not have to scour the Microsoft docs to figure out if documentation exists.

Additionally, without your comment, I would ask myself "Why do I need to find if this has any public documentation?"  I bet many other developer will ask the same thing too.

File: CONTRIBUTING.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (dhoehna) [reply CONTRIBUTING.md]: I'm missing some crucial information.  Why do we need to make a private wrapper?

**Review by krschau — APPROVED** [review]

**Review by nieubank — APPROVED** [review]

**Comment by krschau** [comment]
Adding do-not-merge label as @joadoumie still wants to incorporate some of @dhoehna's feedback before merging.

**Pull request merged**